assets/src/hooks/useSetUserSettingGQL.js

```diff
@@ -0,0 +1,31 @@
+import { gql } from 'apollo-boost'
+import { useMutation } from '@apollo/react-hooks'
+import debounce from 'lodash.debounce'
+
+const UPDATE_USER_SETTING = gql`
+  mutation setUserDefaultSelection($input: UserDefaultSelectionInput!) {
+    setUserDefaultSelection(data: $input) {
+      userDefaultSelection {
+        courseId,
+        defaultViewType,
+        defaultViewValue,
+      }
+    }
+  }
+`
+
+const useSetUserSettingGQL = () => {
+  const [
+    updateUserSetting,
+    { loading: mutationLoading, error: mutationError }
+  ] = useMutation(UPDATE_USER_SETTING)
+
+  const debouncedUpdateUserSetting = debounce(updateUserSetting, 500, {
```

Comment:
I tried to implement debounching here to address this issue @pushyamig, but I don't think it's working correctly. This is a bit of a tricky problem though - it's hard to know exactly when a user is done typing into the input.

Reply:
@justin0022 I take that back that it making a graphql call on focus to the textbox, But when the user starts typing a number it is `making 2 calls instead of one time`.  One with value `{}` and another with value `{6}}. Maybe I got tricked to believe that one is happening on focus and another on typing of the first number. But I think making 2 calls when users start typing is fixable. is my assumption correct? This is the screenshot from it
![Screen Shot 2020-01-20 at 1 43 41 PM](https://user-images.githubusercontent.com/8579775/72751002-5ffb8200-3b8c-11ea-8d5e-8e088c81876f.png)
![Screen Shot 2020-01-20 at 1 43 51 PM](https://user-images.githubusercontent.com/8579775/72751009-65f16300-3b8c-11ea-92c2-0142c805e28e.png)

Reply:
I think I've fixed this issue now, there should just be one request made per input change.